### PR TITLE
Implement debug command

### PR DESCRIPTION
**Pull request by slavchev**
ping @Fatme @teobugslayer

### CODE REVIEW

File: lib/commands/debug.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import helpers = require("./../common/helpers");
+import util = require("util")
+
+export class DebugCommand implements ICommand {
+    constructor(private $platformService: IPlatformService,
+        private $platformCommandParameter: ICommandParameter) { }
+
+    execute(args: string[]): IFuture<void> {
+        return (() => {
+            //
```
Comment (teobugslayer) [inline lib/commands/debug.ts]: you can remove the wait() and starting the new fiber

File: lib/definitions/open.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+declare module "open" {
```
Comment (Fatme) [inline lib/definitions/open.d.ts]: Shouldn't this be in common lib?

File: lib/opener.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+///<reference path=".d.ts"/>
```
Comment (teobugslayer) [inline lib/opener.ts]: This code is the same in AB & NS CLIs. Consider moving it in the common lib and refer it from there

File: lib/services/platform-service.ts [hunk 4]
```diff
@@ -177,13 +177,62 @@ export class PlatformService implements IPlatformService {
 			platform = platform.toLowerCase();
 
 			this.preparePlatform(platform).wait();
-			if(options.emulator) {
+            if (options.emulator) {
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: unneeded change

File: lib/services/platform-service.ts [hunk 5]
```diff
@@ -177,13 +177,62 @@ export class PlatformService implements IPlatformService {
...
 				this.deployOnEmulator(platform).wait();
 			} else {
 				this.deployOnDevice(platform).wait();
 			}
 		}).future<void>()();
-	}
+    }
+
+    public debugPlatform(platform: string): IFuture<void> {
+        return (() => {
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: You can remove starting new fiber and directly return the results of called functions.

File: lib/services/platform-service.ts [hunk 6]
```diff
@@ -177,13 +177,62 @@ export class PlatformService implements IPlatformService {
...
+                this.debugOnDevice(platform).wait();
+            }
+        }).future<void>()();
+    }
+
+    public debugOnEmulator(platform: string): IFuture<void> {
+        return (() => {
+
+            // a bit redundant
+            this.deployOnEmulator(platform).wait();
```
Comment (teobugslayer) [inline lib/services/platform-service.ts]: a bit too peculiar comment. What do you mean?

File: resources/help.txt [hunk 7]
```diff
@@ -235,6 +235,40 @@ Options:
...
+    $ nativescript debug <Platform> [--device <Device ID>] [--debug-brk | --start | --stop | --get-port] [--debug-port <port>]
+
+Platform-specific usage:
+    $ tns debug android --get-port
+    $ tns debug android --debug-brk
+    $ tns debug ios --debug-brk --debug-port 12345
+    $ nativescript debug android --debug-brk
+    $ nativescript debug ios --start --debug-port 12345
+
+TODO: add proper help content
```
Comment (teobugslayer) [inline resources/help.txt]: Ping @ikoevska for help with the help.

File: resources/help.txt [hunk 8]
```diff
@@ -235,6 +235,40 @@ Options:
...
+Debugs your project on a connected device or in the native emulator, if configured. This is shorthand for prepare, build, and deploy.
+<Device ID> is the index or name of the target device as listed by $ tns list-devices.
+
+Before building for the Android emulator, verify that you have met the following requirements.
+    You have added the file paths to the following directories from the Android SDK to your PATH environment variable.
+        tools
+        platform-tools
+    You have created at least one device with the Android Virtual Device manager.
+
+Before building for the iOS simulator, verify that you have installed the ios-sim npm package.
```
Comment (teobugslayer) [inline resources/help.txt]: This should not be needed. We use ios-sim-portable as a dependency.

File: lib/definitions/open.d.ts (hunk 2 above)
Comment (slavchev) [inline lib/definitions/open.d.ts]: I am clues-less here. Is it appropriate for other CLI's? Just let me know and I will move it.

File: lib/services/platform-service.ts (hunk 6 above)
Comment (slavchev) [inline lib/services/platform-service.ts]: The code needs some refactoring. I tried to reuse logic to deployOn* related code path. In this case deployOnEmulator function. I guess I can split in better granularity and use the bits I need. Currently some of the logic will be performed twice.

File: lib/opener.ts (hunk 3 above)
Comment (slavchev) [inline lib/opener.ts]: Done with https://github.com/telerik/mobile-cli-lib/commit/3680ec1545a8d02633963c2ca221f158ffeca183 and https://github.com/NativeScript/nativescript-cli/commit/aff7ef3e7a1b6276598e12062177dca8e21e340b

File: lib/services/platform-service.ts (hunk 5 above)
Comment (slavchev) [inline lib/services/platform-service.ts]: Done with https://github.com/NativeScript/nativescript-cli/commit/cd1d205dba1fa0bea6d7d158c385306e4cd64cf6

**Comment by slavchev** [comment]
ping @ErjanGavalji

**Comment by ErjanGavalji** [comment]
:+1:

**Pull request merged**